deploy-tool/main.go

```diff
@@ -85,17 +85,34 @@ type TerraformOutput struct {
 	} `json:"specific_manifests"`
 }
 
+type privateKeyMarshaler func(*ecdsa.PrivateKey) ([]byte, error)
+
+// marshalX962UncompresedPrivateKey encodes an ECDSA P256 private key into the
+// format expected by libprio-rs encrypt::PrivateKey, which is the X9.62
+// uncompressed public key concatenated with the secret scalar.
+func marshalX962UncompressedPrivateKey(ecdsaKey *ecdsa.PrivateKey) ([]byte, error) {
+	marshaledPublicKey := elliptic.Marshal(elliptic.P256(), ecdsaKey.PublicKey.X, ecdsaKey.PublicKey.Y)
+	return append(marshaledPublicKey, ecdsaKey.D.Bytes()...), nil
+}
+
+// marshakPKCS8PrivateKey encodes an ECDSA P256 private key into a PKCS#8
+// document. This function adapts x509.MarshalPKCS8PrivateKey to the
```

Comment:
I don't think an adapter is needed; you can just pass `x509.MarshalPKCS8PrivateKey` directly.

Reply:
I cannot:

Reply:
Ohhh I see, it takes a blank interface! That makes sense. I'm fine with this as it stands.

You could also change the `privateKeyMarshaler` and `marshalX962UncompresedPrivateKey` types to take a blank interface, which would save you having to make a wrapper. But I don't think it's necessary.

Reply:
...and if I change `privateKeyMarshaler` to take `interface{}` to match `x509.MarshalPKCS8PrivateKey`, then I would have to do a type check and cast in `marshalX962UncompressedPrivateKey` which I find uglier than providing the adapter for `x509.MarshalPKCS8PrivateKey`.